framework/src/bundle/BundlePrivate.cpp

```diff
@@ -420,6 +420,10 @@ namespace cppmicroservices
     {
         auto l = this->Lock();
         US_UNUSED(l);
+        if(coreCtx->FrameworkStopped()){
```

Comment:
there is still a race when checking the state of the framework since the return value of `stopped` could change after this function returns but before the if statement is evaluated.